src/vm/arm64/stubs.cpp

```diff
@@ -1313,28 +1314,52 @@ LONG CLRNoCatchHandler(EXCEPTION_POINTERS* pExceptionInfo, PVOID pv)
     return EXCEPTION_CONTINUE_SEARCH;
 }
 
+#ifndef CROSSGEN_COMPILE
 void StompWriteBarrierEphemeral(bool isRuntimeSuspended)
 {
+    _ASSERTE(isRuntimeSuspended);
+
     JIT_UpdateWriteBarrierState(GCHeapUtilities::IsServerHeap());
 }
 
 void StompWriteBarrierResize(bool isRuntimeSuspended, bool bReqUpperBoundsCheck)
 {
+    // The runtime is not always suspended when this is called.  If a thread was
+    // preempted while holding the old card_table ptr in a register, when
+    // resumes it will write to the old card_table (bad).  Same is true for
+    // card_bundle_table
+    //
+    // Suspend the runtime to avoid this race condition
```

Comment:
Suspending of the runtime is always way more expensive than FlushProcessWriteBuffers. I do not think the runtime should be suspended here.